server/issue_handler.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package server
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/mattermost/mattermost-mattermod/model"
+	"github.com/mattermost/mattermost-server/v5/mlog"
+)
+
+func (s *Server) issueEventHandler(w http.ResponseWriter, r *http.Request) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultRequestTimeout*time.Second)
+	defer cancel()
+
+	event, err := PullRequestEventFromJSON(r.Body)
+	if err != nil {
+		mlog.Error("could not parse issue event", mlog.Err(err))
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
This is just a general thing and not related to this PR, but I note that in case of error scenarios, we are just writing the status code and returning. No body is written. Wondering if it makes sense to use the `http.Error` function to write out the error in the response along with the status code? That will allow us to view the error response in the GitHub console too.